bot/src/main/kotlin/io/heapy/kotbot/bot/rule/Rules.kt

```diff
@@ -156,18 +157,23 @@ class DeleteVoiceMessageRule : Rule {
 }
 
 class CombotCasRule(
-    private val client: HttpClient
+    private val client: HttpClient,
+    private val casConfiguration: CasConfiguration
 ) : Rule {
     override fun validate(update: Update): Flow<Action> = flow {
         update.anyMessage?.let { message ->
             val userId = message.from.id
-            val response = client.get<CasResponse>("https://api.cas.chat/check?user_id=$userId")
-            if (response.ok) {
-                LOGGER.info("User $userId is CAS banned")
-                emit(DeleteMessageAction(message))
-                emit(KickUserAction(message.chatId, userId))
+            if (!casConfiguration.allowlist.contains(userId.toLong())) {
+                val response = client.get<CasResponse>("https://api.cas.chat/check?user_id=$userId")
+                if (response.ok) {
+                    LOGGER.info("User $userId is CAS banned")
+                    emit(DeleteMessageAction(message))
+                    emit(KickUserAction(message.chatId, userId))
+                } else {
+                    LOGGER.info("User $userId is NOT CAS banned")
```

Comment:
Won't this flood your logs? Or that's okay?

Reply:
Log volume here is pretty small. And even before one log.info happens on each message. I even thinking to increase log volume, and put more metadata about user to logs :)

Reply:
I want to try something small and self-hosted as [loki](https://github.com/grafana/loki) for log storing and processing. Then it's will be even lesser issue

Reply:
Sure, if that works for you -- no problem :)